src/index.ts

```diff
@@ -859,10 +859,16 @@ export class ValidationService {
                     });
                 form.dispatchEvent(validationEvent);
 
-                this.handleValidated(form, success, e);
+                // Firefox fix: redispatch 'submit' after finished handling this event
+                setTimeout(() => {
+                    try {
+                        this.handleValidated(form, success, e);
+                    } finally {
+                        validating = false;
```

Comment:
I'm not thrilled with this being pulled out of the `finally()`. Might try promisifying the `setTimeout()`.